### ISSUE TITLE
Support for T5 Architecture

### ISSUE THREAD

**Issue by niranjanakella** [EricLBuehler/mistral.rs#384]
Hello @EricLBuehler, opening this issue as part of T5 architecture support in mistral.rs. 

Relates to: #156

**Comment by EricLBuehler** [comment]
Hi @niranjanakella!

Thank you for opening this issue. Just to clarify, would this be a quantized or nonquantized implementation?

**Comment by niranjanakella** [comment]
@EricLBuehler Non-Quantized f16,32 implementation currently holds more precedence. But if possible, would also like to have a quantized implementation too.

**Comment by EricLBuehler** [comment]
> Non-Quantized f16,32 implementation currently holds more precedence. But if possible, would also like to have a quantized implementation too.

Sounds great, I'll get started on an implementation.

> Also I wish to know if LoRA adapters can be loaded at runtime without merging them into the model. It would be a huge game changer for most applications given the fact that many developers train multiple adapters. Would be great to attach multiple adapters during runtime.

We actually have this feature already! You can activate adapters at runtime by preloading some and then use per-request adapter specification to have granular control. Docs: https://github.com/EricLBuehler/mistral.rs/blob/master/docs/ADAPTER_MODELS.md#adapter-model-dynamic-adapter-activation.

**Comment by EricLBuehler** [comment]
Hi @niranjanakella! Sorry for the delay; I have been busy with the Idefics 2 implementation (#309). I should have a prototype ready tonight, though!

**Comment by niranjanakella** [comment]
@EricLBuehler No problem sounds good. I am looking forward to trying it out soon.

**Comment by EricLBuehler** [comment]
See: #432.

**Comment by cyanic-selkie** [comment]
Hi, is there any news for this? I have the exact same use case, albeit with a quantized model.
